salt/modules/win_pkg.py

```diff
@@ -426,80 +426,252 @@ def _get_reg_software():
     keys, it will return a dict with the display name as the key and the
     version as the value
     '''
-    ignore_list = ['AddressBook',
-                   'Connection Manager',
-                   'DirectDrawEx',
-                   'Fontcore',
-                   'IE40',
-                   'IE4Data',
-                   'IE5BAKEX',
-                   'IEData',
-                   'MobileOptionPack',
-                   'SchedulingAgent',
-                   'WIC',
-                   'Not Found',
-                   '(value not set)',
-                   '',
-                   None]
-
+    # Logic for this can be found in this question:
+    # https://social.technet.microsoft.com/Forums/windows/en-US/d913471a-d7fb-448d-869b-da9025dcc943/where-does-addremove-programs-get-its-information-from-in-the-registry
     reg_software = {}
-    hive = 'HKLM'
-    key = "Software\\Microsoft\\Windows\\CurrentVersion\\Uninstall"
-
-    def update(hive, key, reg_key, use_32bit):
-        # 2018 this code has been update to reflect some of utils/pkg/win.py logic
-        # i.e. check_reg, and checking of DisplayName and DisplayVersion.
-        d_name = ''
-        d_vers = ''
-
-        d_name_regdata = __salt__['reg.read_value'](hive,
-                                                    '{0}\\{1}'.format(key, reg_key),
-                                                    'DisplayName',
-                                                    use_32bit)
+
+    def skip_component(hive, key, reg_key, use_32bit):
+        '''
+        Returns:
+            bool: True if the component needs to be skipped, otherwise False
+        '''
+        # Skip SystemComponents
+        # Skip WindowsInstaller (removes Compiler for Python 2.7)
+        #for check in ['SystemComponent', 'WindowsInstaller']:
```

Comment:
WindowsInstaller means its an MSI.

Reply:
Yes. Added a check for the GUID in HKLM:\Software\Classes\Installer\Products\. If it's there, then it will be added. Added some code comments about that as well.